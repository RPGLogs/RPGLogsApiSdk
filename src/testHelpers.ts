import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
