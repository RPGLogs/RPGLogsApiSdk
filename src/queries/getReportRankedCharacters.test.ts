import { buildTestSdk } from '../testHelpers';

test('returns data', async () => {
  const response = await buildTestSdk().getReportRankedCharacters({
    code: 'pJTWtB8Yj4F1rc9Q',
    includeServer: true,
    includeGuilds: true,
    includeGameData: false,
  });

  const dataToSnapshot = response?.reportData?.report?.rankedCharacters?.map(
    (x) => ({
      id: x?.id,
      name: x?.name,
      classID: x?.classID,
    })
  );

  expect(dataToSnapshot).toMatchSnapshot();
});
