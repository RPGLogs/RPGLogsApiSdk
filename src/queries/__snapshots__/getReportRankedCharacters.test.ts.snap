// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
Array [
  Object {
    "classID": 5,
    "id": 1039279,
    "name": "Heaat",
  },
  Object {
    "classID": 9,
    "id": 19415756,
    "name": "Centarios",
  },
  Object {
    "classID": 12,
    "id": 31557524,
    "name": "Màrz",
  },
  Object {
    "classID": 10,
    "id": 34288357,
    "name": "Xenubis",
  },
  Object {
    "classID": 10,
    "id": 34288364,
    "name": "Dissident",
  },
  Object {
    "classID": 9,
    "id": 34317966,
    "name": "Gärth",
  },
  Object {
    "classID": 7,
    "id": 34557854,
    "name": "Lambkorma",
  },
  Object {
    "classID": 4,
    "id": 35012063,
    "name": "Poorsorcero",
  },
  Object {
    "classID": 9,
    "id": 35563395,
    "name": "Mangochutney",
  },
  Object {
    "classID": 4,
    "id": 35774109,
    "name": "Mercurymage",
  },
  Object {
    "classID": 5,
    "id": 37462621,
    "name": "Pandamanium",
  },
  Object {
    "classID": 2,
    "id": 37491830,
    "name": "Flucloxx",
  },
  Object {
    "classID": 3,
    "id": 37962356,
    "name": "Rustedarrow",
  },
  Object {
    "classID": 11,
    "id": 40067043,
    "name": "Flyxw",
  },
  Object {
    "classID": 2,
    "id": 40483585,
    "name": "Betwixt",
  },
  Object {
    "classID": 10,
    "id": 40777050,
    "name": "Nògg",
  },
  Object {
    "classID": 12,
    "id": 46159040,
    "name": "Xyä",
  },
  Object {
    "classID": 7,
    "id": 51175867,
    "name": "Scárý",
  },
  Object {
    "classID": 9,
    "id": 53075143,
    "name": "Antakka",
  },
  Object {
    "classID": 1,
    "id": 53801762,
    "name": "Heamx",
  },
  Object {
    "classID": 6,
    "id": 56286106,
    "name": "Éxory",
  },
  Object {
    "classID": 7,
    "id": 59060651,
    "name": "Dragobbi",
  },
  Object {
    "classID": 3,
    "id": 59991946,
    "name": "Matreira",
  },
]
`;
